app/controllers/cclow/geography/climate_targets_controller.rb

```diff
@@ -5,20 +5,46 @@ class ClimateTargetsController < CCLOWController
 
       def index
         add_breadcrumb('Climate targets', request.path)
-        @ndc_targets = if @geography.eu_member?
-                         eu = ::Geography.where(iso: 'EUR').first
-                         eu.targets.published.where(source: 'ndc')
-                       else
-                         @geography.targets.published.where(source: 'ndc')
-                       end
-        @climate_targets = @geography.targets.published.where.not(source: 'ndc')
+        @list = targets_list_by_sector
       end
 
       def show
         @target = ::Target.find(params[:id])
         add_breadcrumb('Climate targets', cclow_geography_climate_targets_path(@geography))
         add_breadcrumb(@target.id, request.path)
       end
+
+      def laws_sector
+        add_breadcrumb('Climate targets', request.path)
+        @sector = LawsSector.find_by(name: params[:law_sector])
+        @ndc_targets = if @geography.eu_member?
+                         eu = ::Geography.where(iso: 'EUR').first
+                         eu.targets.published.where(source: 'ndc', sector: @sector)
+                       else
+                         @geography.targets.published.where(source: 'ndc', sector: @sector)
+                       end
+        @ndc_targets = CCLOW::TargetDecorator.decorate_collection(@ndc_targets)
+
+        @climate_targets = @geography.targets.published.where(sector: @sector).where.not(source: 'ndc')
+        @climate_targets = CCLOW::TargetDecorator.decorate_collection(@climate_targets)
+        render 'cclow/geography/climate_targets/target_sector'
+      end
+
+      private
+
+      def targets_list_by_sector
```

Comment:
I think this should be a method inside of Geography, as it's quite specific to a single geography. And instead of using ruby we could query the results directly with something like:

`LawsSector.joins(:targets).where(targets: { geography_id: @geography}).select('laws_sectors.name, laws_sectors.id, targets.source, COUNT(*) AS count').group('laws_sectors.id', 'laws_sectors.name', 'targets.source')`, that would return the sector names and the count and then we can just iterate over it. What do you think?

Reply:
hmmm, I think that tentative query would still need some work, but would be achievable =)